manifests/common/ceilometer.pp

```diff
@@ -3,10 +3,21 @@
 class openstack::common::ceilometer {
   $is_controller = $::openstack::profile::base::is_controller
 
+  $ceilometer_management_address = $::openstack::config::ceilometer_address_management
   $controller_management_address = $::openstack::config::controller_address_management
 
+  $mongo_username = $::openstack::config::ceilometer_mongo_username
   $mongo_password = $::openstack::config::ceilometer_mongo_password
-  $mongo_connection = "mongodb://${controller_management_address}:27017/ceilometer"
+
+  if ! $mongo_username {
+    fail('mongo_username must be set when ceilometer is being configured')
```

Comment:
I would prefer not to hard fail, as this seemed to be working correctly before we were setting the username and password. I think it should use the new connection string if mongo_username and mongo_password are set, and use the old one if they are undef.